File: data/cpp/tests/test_tenacity_stop_after_attempt.cpp
#include <gtest/gtest.h>
#include "../src/function_tenacity_stop_after_attempt.cpp"

class TestMightFailFunction : public ::testing::Test {
protected:
    RetryFunction func;
};

TEST_F(TestMightFailFunction, TestSuccessOnFirstTry) {
    int result = func.might_fail_function(6);
    EXPECT_EQ(result, 6);
}

TEST_F(TestMightFailFunction, TestSuccessAfterRetry) {
    int result = func.might_fail_function(5);
    EXPECT_EQ(result, 5);
}

TEST_F(TestMightFailFunction, TestNoRetryOnTooSmallValue) {
    EXPECT_THROW({
        func.might_fail_function(1, 0, false);
    }, std::runtime_error);
}

TEST_F(TestMightFailFunction, TestRuntimeErrorOnTooSmallValue) {
    EXPECT_THROW({
        func.might_fail_function(1, 0, true);
    }, std::runtime_error);
}

TEST_F(TestMightFailFunction, TestRuntimeErrorWithExtraSleep) {
    EXPECT_THROW({
        func.might_fail_function(4, 0.5, true);
    }, std::runtime_error);
}

int main(int argc, char **argv) {
    ::testing::InitGoogleTest(&argc, argv);
    return RUN_ALL_TESTS();
}

File: data/java/tests/test_tenacity_stop_after_attempt.java
// 测试代码保持不变
package com.example;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class test_tenacity_stop_after_attempt {
    private function_tenacity_stop_after_attempt func;

    @BeforeEach
    void setUp() {
        func = new function_tenacity_stop_after_attempt();
    }

    @Test
    void test_success_on_first_try() {
        int result = func.might_fail_function(6, 0.0f, false);
        assertEquals(6, result);
    }

    @Test
    void test_success_after_retry() {
        int result = func.might_fail_function(5, 0.0f, false);
        assertEquals(5, result);
    }

    @Test
    void test_no_retry_on_too_small_value() {
        assertThrows(RuntimeException.class, () ->
                func.might_fail_function(1, 0.0f, false)
        );
    }

    @Test
    void test_runtimeerror_on_too_small_value() {
        RuntimeException exception = assertThrows(RuntimeException.class, () ->
                func.might_fail_function(1, 0.0f, true)
        );
        assertTrue(exception.getMessage().contains("Operation failed after retries"));
    }

    @Test
    void test_runtimeerror_with_extra_sleep() {
        RuntimeException exception = assertThrows(RuntimeException.class, () ->
                func.might_fail_function(4, 0.5f, true)
        );
        assertTrue(exception.getMessage().contains("Operation failed after retries"));
    }
}